rust/main/agents/relayer/src/msg/pending_message.rs

```diff
@@ -516,61 +519,90 @@ impl PendingOperation for PendingMessage {
+    /// If the message has been retried more than `retries_before_skipping`, it will return `None`.
     /// In case of failure, behaves like `Self::new(...)`.
-    pub fn from_persisted_retries(
+    pub fn maybe_from_persisted_retries(
         message: HyperlaneMessage,
         ctx: Arc<MessageContext>,
         app_context: Option<String>,
-    ) -> Self {
-        // Attempt to fetch status about message from database
-        let message_status = match ctx.origin_db.retrieve_status_by_message_id(&message.id()) {
-            Ok(Some(status)) => {
-                // This event is used for E2E tests to ensure message statuses
-                // are being properly loaded from the db
-                tracing::event!(
-                    if cfg!(feature = "test-utils") {
-                        Level::DEBUG
-                    } else {
-                        Level::TRACE
-                    },
-                    ?status,
-                    id=?message.id(),
-                    RETRIEVED_MESSAGE_LOG,
-                );
-                status
-            }
-            _ => {
-                tracing::event!(
-                    if cfg!(feature = "test-utils") {
-                        Level::DEBUG
-                    } else {
-                        Level::TRACE
-                    },
-                    "Message status not found in db"
-                );
-                PendingOperationStatus::FirstPrepareAttempt
-            }
-        };
+        max_message_retries: u32,
+    ) -> Option<Self> {
+        let num_retries =
+            Self::get_retries_or_skip(ctx.origin_db.clone(), &message, max_message_retries)?;
+        let message_status = Self::get_message_status(ctx.origin_db.clone(), &message);
+        let mut pending_message = Self::new(
+            message,
+            ctx,
+            message_status,
+            app_context,
+            max_message_retries,
+        );
+        if num_retries > 0 {
+            let next_attempt_after =
+                PendingMessage::calculate_msg_backoff(num_retries, max_message_retries, None)
+                    .map(|dur| Instant::now() + dur);
+            pending_message.num_retries = num_retries;
+            pending_message.next_attempt_after = next_attempt_after;
+        }
+        Some(pending_message)
+    }
 
-        let num_retries = match ctx
-            .origin_db
-            .retrieve_pending_message_retry_count_by_message_id(&message.id())
-        {
+    fn get_retries_or_skip(
+        origin_db: Arc<dyn HyperlaneDb>,
+        message: &HyperlaneMessage,
+        max_message_retries: u32,
+    ) -> Option<u32> {
+        let num_retries = Self::get_num_retries(origin_db.clone(), message);
+        if Self::should_skip(num_retries, max_message_retries) {
+            return None;
+        }
+        Some(num_retries)
+    }
+
+    pub fn should_skip(retry_count: u32, retries_before_skipping: u32) -> bool {
```

Comment:
nit: any reason to have a different name for `retries_before_skipping` vs elsewhere? just imagine consistency within `PendingMessage` is easier to track

Reply:
fixed